libtiledbsoma/src/soma/soma_binary_column.cc

```diff
@@ -0,0 +1,404 @@
+
+        return std::make_shared<SOMABinaryColumn>(dimension);
+
+    } else {
+        auto attribute = ArrowAdapter::tiledb_attribute_from_arrow_schema(
+            ctx, schema, type_metadata, platform_config);
+
+        return std::make_shared<SOMABinaryColumn>(
+            SOMABinaryColumn(attribute.first, attribute.second));
+    }
+}
+
+void SOMABinaryColumn::_set_dim_points(
+    ManagedQuery& query, const std::any& points) const {
+    if (!isIndexColumn()) {
+        throw TileDBSOMAError(fmt::format(
+            "[SOMABinaryColumn][_set_dim_points] Column with name {} is not an "
+            "index "
+            "column",
+            name()));
+    }
+
+    std::vector<std::string> casted_points;
+
+    for (const auto& bytes :
+         std::any_cast<std::span<const std::vector<std::byte>>>(points)) {
+        casted_points.push_back(std::string(
+            reinterpret_cast<const char*>(bytes.data()), bytes.size()));
+    }
+
+    query.select_points(name(), casted_points);
+}
+
+void SOMABinaryColumn::_set_dim_ranges(
+    ManagedQuery& query, const std::any& ranges) const {
+    if (!isIndexColumn()) {
+        throw TileDBSOMAError(fmt::format(
+            "[SOMABinaryColumn][_set_dim_ranges] Column with name {} is not an "
+            "index "
+            "column",
+            name()));
+    }
+
+    std::vector<std::pair<std::string, std::string>> casted_ranges;
+
+    for (const auto& bytes : std::any_cast<std::vector<
+             std::pair<std::vector<std::byte>, std::vector<std::byte>>>>(
+             ranges)) {
+        casted_ranges.push_back(std::make_pair(
+            std::string(
+                reinterpret_cast<const char*>(bytes.first.data()),
+                bytes.first.size()),
+            std::string(
+                reinterpret_cast<const char*>(bytes.second.data()),
+                bytes.second.size())));
+    }
+
+    query.select_ranges(name(), casted_ranges);
+}
+
+void SOMABinaryColumn::_set_current_domain_slot(
+    NDRectangle& rectangle, std::span<const std::any> domain) const {
+    if (!isIndexColumn()) {
+        throw TileDBSOMAError(fmt::format(
+            "[SOMABinaryColumn][_set_current_domain_slot] Column with name {} "
+            "is "
+            "not "
+            "an index column",
+            name()));
+    }
+
+    if (domain.size() != 1) {
+        throw TileDBSOMAError(fmt::format(
+            "[SOMABinaryColumn][_set_current_domain_slot] Invalid domain size. "
+            "Expected 1, got {}",
+            domain.size()));
+    }
+
+    auto dom = std::any_cast<std::array<std::vector<std::byte>, 2>>(domain[0]);
+    if (dom[0].size() == 0 && dom[1].size() == 0) {
+        rectangle.set_range(name(), "", "\x7f");
+    } else {
+        throw TileDBSOMAError(fmt::format(
+            "[SOMABinaryColumn][_set_current_domain_slot] domain (\"{}\", "
+            "\"{}\") cannot be set for "
+            "binary index columns: please use "
```

Comment:
```suggestion
            "\"{}\") cannot be set for binary index columns: please use "
```

or wherever `clang-format` puts the new linebreaks. Regardless, this is in too many pieces. (`clang-format` will split long lines but not join short ones -- so we need to do the latter manually.)